### DISCUSSION TITLE
PostgresQL error `no schema has been selected to create in`

### DISCUSSION THREAD

**Discussion by DenuxPlays** [SeaQL/sea-orm#2021] [General]
## Description

I created a few tables in my database and let the sea-orm-cli generate Rust types from the tables.
Now I want to implement that the schema gets loaded when the tables do not exist in the database.
But now I get the following error:
![image](https://github.com/SeaQL/sea-orm/assets/83671398/eedca054-3db0-4b0b-b71b-036824cbb7ce)

## Steps to Reproduce

1. set up postgresql docker container
2. create a few tables
3. let the cli generate rust types from them
4. delete all tables from the database
5. now let the orm create them (using: `schema.create_table_from_entity(user::Entity)`)

### Expected Behavior

That the database uses the standard schema (default public)

### Actual Behavior

Getting an SQLx error.

### Reproduces How Often

Yes always.
Even after deleting the data folder etc.

### Workarounds

Adding this to the table: `#[sea_orm(table_name = "user", schema_name = "public")]`
Doesn't fix migrations tho!!!

## Reproducible Example

**Will do this tomorrow!!!**

## Versions

![image](https://github.com/SeaQL/sea-orm/assets/83671398/40c301ca-4fac-4ef6-a910-d3dcb3319359)

Database: PostgresQL version 16.1 (Docker)
OS on Docker: Alpine v3.18
Host Os: Fedora 39 Workstation

**Comment by DenuxPlays** [comment]
Tried giving the user all usage and create permissions on the schema and defining the schema to use in the database uri but both don't work.

**Comment by tyt2y3** [comment]
I don't understand the problem, did you put the schema name on the connection string? i.e. https://www.sea-ql.org/SeaORM/docs/install-and-config/connection/#postgres

**Reply by DenuxPlays** [reply]
Yes I did.
Still the same.
Also checked permissions on the schema...

**Comment by DenuxPlays** [comment]
Fixed it!

In the ConnectOptions I had set `set_schema_search_path` to my database name instead of my database schema.
Small typo made a big difference. :/
